app/src/test/kotlin/io/intrepid/skotlinton/utils/LiveDataObserverTest.kt

```diff
@@ -0,0 +1,85 @@
+package io.intrepid.skotlinton.utils
+
+import android.view.View
+import android.widget.TextView
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.LifecycleRegistry
+import androidx.lifecycle.MutableLiveData
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import io.intrepid.skotlinton.testutils.LiveDataTestBase
+import org.amshove.kluent.shouldEqual
+import org.junit.Before
+import org.junit.Test
+
+class LiveDataObserverTest : LiveDataTestBase() {
+    private val lifecycleOwner = mock<LifecycleOwner>()
+    private val mockObserver = MockObserver(lifecycleOwner)
+    private val lifecycleRegistry = LifecycleRegistry(lifecycleOwner)
+
+    @Before
+    fun setup() {
+        whenever(lifecycleOwner.lifecycle).thenReturn(lifecycleRegistry)
+        lifecycleRegistry.markState(Lifecycle.State.STARTED)
+    }
+
+    @Test
+    fun observe() {
+        var observedValue: String? = null
+
+        val liveData = MutableLiveData<String>()
+        mockObserver.apply {
```

Comment:
I more often find myself using `run` than `apply` for this sort of scoping. But not sure which one we want to prescribe as a code style or best practice.

Reply:
I don't have a strong preference on either one. There's also `with(mockObserver{})`. Not sure if there is a general Kotlin guideline on this.

Reply:
I wrote a Medium blog post about this :)

https://medium.com/@streetsofboston/lets-also-apply-a-run-with-kotlin-on-our-minds-56f12eaef5e3?source=friends_link&sk=5a687e82e6339aa74846b5bed33757de

Reply:
so it sounds like `with(mockObserver){ }` is the preferable approach here since we don't care about the return. Am I interpreting the blog correctly?

Reply:
Actually there's a page on this now in the official kotlin docs too: https://kotlinlang.org/docs/reference/scope-functions.html

My read of the [function selection table](https://kotlinlang.org/docs/reference/scope-functions.html#function-selection) is that `apply` is better than `run`, though `with` would work as well. It's a little funky because we're scope-changing just to get access to an extension function, which is not a case the docs explicitly call out.

Reply:
We can examine the usage in Example2Fragment's onViewCreated. The test version will just follow that.

Reply:
We can examine the usage in Example2Fragment's onViewCreated. The test version will just follow that.